fix compile fails with asan and -Wredundant-decls

hellowangsai opened the pull request:
If compile with asan, unaligned_access.h redeclares the same sanitizer
function with sanitizer/common_interface_defs.h.

This patch removes these declarations, and includes sanitizer header
file to avoid this warning.

hellowangsai commented:
@googlebot I signed it!

suertreus commented:
googlebot is having trouble associating the CLA you signed with the email under which the commit was committed (@smartx.com).  Can you please adjust the commit or (if possible) sign the CLA with that account too?

Here's the text from https://cla.developers.google.com/ in case it does a better job explaining than I have:
> This page says I signed the CLA, but I was notified that a CLA couldn't be found.
> Ensure you submitted your contribution using one of the email addresses associated with the account you're signed in to and/or using the GitHub username you provided us with when you signed the CLA. This is how we match your submission with your CLA.

hellowangsai commented:
> @googlebot I signed it!

@googlebot I signed it!

hellowangsai committed:
fix compile fails with asan and -Wredundant-decls

If compile with asan, unaligned_access.h redeclares the same sanitizer
function with sanitizer/common_interface_defs.h.

This patch removes these declarations and includes sanitizer header
file to avoid this warning.

S3od22 commented:
ا

S3od22 reviewed (commented):
ا
